strax/processing/hitlets.py

```diff
@@ -0,0 +1,526 @@
+    
+    Notes:
+        Assumes new_hitlets to be sorted in time.
+    """
+    for ind, hit in enumerate(new_hitlets):
+
+        updated = False
+        counter = 0
+        current_ri = hit['record_i']
+        while not updated:
+            r = records[current_ri]
+            # Hitlet must only partially be contained in record_i:
+            time = hit['time']
+            end_time = strax.endtime(hit)
+            start_in = (r['time'] <= time) & (time < strax.endtime(r))
+            end_in = (r['time'] < end_time) & (end_time <= strax.endtime(r))
+            if start_in or end_in:
+                hit['record_i'] = current_ri
+                break
+            else:
+                last_ri = current_ri
+                current_ri = next_ri[current_ri]
+                counter += 1
+                
+            if current_ri == -1:
+                print('Record:\n', r, '\nHit:\n', hit)
+                raise ValueError('Was not able to find record_i')
+
+            if counter > TRIAL_COUNTER_NEIGHBORING_RECORDS:
+                print(ind, last_ri)
+                raise RuntimeError('Tried too often to find correct record_i.')
+
+
+# ----------------------
+# Hitlet properties:
+# ----------------------
+@export
+@numba.njit(cache=True, nogil=True)
+def hitlet_properties(hitlets):
+    """
+    Computes additional hitlet properties such as amplitude, FHWM, etc.
+    """
+    for h in hitlets:
+
+        dt = h['dt']
+        data = h['data'][:h['length']]
+        # Compute amplitude
+        amp_ind = np.argmax(data)
+        amp_time = int(amp_ind * dt)
+        height = data[amp_ind]
+
+        h['amplitude'] = height
+        h['time_amplitude'] = amp_time
+
+        # Computing FWHM:
+        left_edge, right_edge = get_fwxm(h, 0.5)
+        width = right_edge - left_edge
+
+        # Computing FWTM:
+        left_edge_low, right_edge = get_fwxm(h, 0.1)
+        width_low = right_edge - left_edge_low
+
+        h['fwhm'] = width
+        h['left'] = left_edge
+        h['low_left'] = left_edge_low
+        h['fwtm'] = width_low
+
+
+NO_FWXM = -42
+@numba.njit(cache=True, nogil=True)
+def get_fwxm(hitlet, fraction=0.5):
+    """
+    Estimates the left and right edge of a specific height percentage.
+
+    :param hitlet: Single hitlet
+    :param fraction: Level for which the width shall be computed.
+    :returns: Two floats, left edge and right edge in ns
+
+    Notes:
+        The function searches for the last sample below and above the
+        specified height level on the left and right hand side of the
+        maximum. When the samples are found the width is estimated based
+        upon a linear interpolation between the respective samples. In
+        case, that the samples cannot be found for either one of the
+        sides the corresponding outer most bin edges are used: left 0;
+        right last sample + 1.
+    """
+    data = hitlet['data'][:hitlet['length']]
+    max_val = hitlet['amplitude'] * fraction
```

Comment:
For peaks this would be area right? perhaps we can make it an kwarg such that it's also sable for peaks

Reply:
Okay no just checked peaks do not have a "height" field.